timeseries/blueflood/blueflood_test.go

```diff
@@ -16,607 +16,240 @@ package blueflood
+	Resolution:     30 * time.Second,
+	FirstAvailable: 0,
+	// TimeToLive is open, so setting it to 1 day exactly means queries from -1d to now will need 5min.
+	TimeToLive: 1*day + 30*time.Second,
+}
+var resolution5Min = Resolution{
+	Name:           "5MIN",
+	Resolution:     5 * time.Minute,
+	FirstAvailable: 1 * day,
+	TimeToLive:     30 * day,
+}
+var resolution60Min = Resolution{
+	Name:           "60MIN",
+	Resolution:     time.Hour,
+	FirstAvailable: 15 * day,
+	TimeToLive:     90 * day,
+}
+var resolution1440Min = Resolution{
+	Name:           "1440MIN",
+	Resolution:     day,
+	FirstAvailable: 180 * day,
+	TimeToLive:     900 * day,
+}
 
-	for _, test := range []struct {
-		name               string
-		metricMap          map[util.GraphiteMetric]api.TaggedMetric
-		queryMetric        api.TaggedMetric
-		sampleMethod       timeseries.SampleMethod
-		timerange          api.Timerange
-		clientConfig       Config
-		queryURL           string
-		queryResponse      string
-		queryResponseCode  int
-		queryDelay         time.Duration
-		expectedErrorCode  timeseries.ErrorCode
-		expectedSeriesList api.Timeseries
-	}{
-		{
-			name: "Success case",
-			queryMetric: api.TaggedMetric{
-				MetricKey: api.MetricKey("some.key"),
-				TagSet:    api.TagSet{"tag": "value"},
-			},
-			sampleMethod: timeseries.SampleMean,
-			timerange:    timerange,
-			queryURL:     defaultQueryURL,
-			clientConfig: defaultClientConfig,
-			queryResponse: `{
-        "unit": "unknown",
-        "values": [
-          {
-            "numPoints": 1,
-            "timestamp": 12000,
-            "average": 5
-          },
-          {
-            "numPoints": 1,
-            "timestamp": 13000,
-            "average": 3
-          }
-        ],
-        "metadata": {
-          "limit": null,
-          "next_href": null,
-          "count": 2,
-          "marker": null
-        }
-      }`,
-			expectedSeriesList: api.Timeseries{
-				Values: []float64{5, 3},
-				TagSet: api.TagSet{"tag": "value"},
+var testResolutions = []Resolution{resolutionFull, resolution5Min, resolution60Min, resolution1440Min}
+
+func TestPlanFetchIntervals(t *testing.T) {
+	// Note: this constant is not completely arbitrary. It has lots of factors,
+	// which means that it lies on a lot of resolution boundaries,
+	// so most resolutions will be able to work without rounding (e.g., 31ms).
+	nowMillis := int64(12331800) * 60000
+	nowFunc := func() time.Time {
+		timeValue := time.Unix(nowMillis/1000, nowMillis%1000*1e6)
+		return timeValue
+	}
+	makeInterval := func(beforeStart time.Duration, beforeEnd time.Duration) api.Interval {
+		return api.Interval{Start: nowFunc().Add(-beforeStart), End: nowFunc().Add(-beforeEnd)}
+	}
+	a := assert.New(t).Contextf("Blueflood planFetchIntervals")
+	type test struct {
+		resolutions []Resolution
+		requested   api.Interval
+		lowerBound  time.Duration
+		expected    map[Resolution]api.Interval
+		error       bool
+	}
+	testcases := []test{
+		{
```

Comment:
Could we add some tests for the following:
* Breaking off inside of `FULL`, like -160days to -6h
* Covering the `1440MIN` resolution since it's an island, like -190days to -185days